signwriting_animation/data/data_loader.py

```diff
@@ -0,0 +1,123 @@
+import os
+import random
+import torch
+import pandas as pd
+from torch.utils.data import Dataset, DataLoader
+from pose_format.torch.masked.collator import zero_pad_collator
+from pose_format.pose import Pose
+from pose_anonymization.data.normalization import normalize_mean_std
+from signwriting.visualizer.visualize import signwriting_to_image
+from transformers import CLIPProcessor
+from PIL import Image
+
+class DynamicPosePredictionDataset(Dataset):
+    """
+    Dataset for dynamic on-the-fly sampling of normalized pose data
+    with SignWriting condition images and scalar metadata padded as length-1 tensors.
+    """
+    def __init__(self,data_dir,csv_path,past_frames=40,future_frames=20,with_metadata=True):
+        super().__init__()
+        self.data_dir = data_dir
+        self.num_past_frames = num_past_frames
+        self.num_future_frames = num_future_frames
```

Comment:
this will obviously not work - yoy are using an undefined variable. the function gets `future_frames` and you use `num_future_frames`. I like `num_future_frames` more, but the function needs to get it, and you need to check your code before submitting a PR